cromwell_tools/_cli.py

```diff
@@ -0,0 +1,61 @@
+import argparse
+
+from cromwell_tools._cromwell_api import CromwellAPI
+from cromwell_tools._cromwell_auth import CromwellAuth
+
+
+def parser(arguments=None):
+    main_parser = argparse.ArgumentParser()
+
+    subparsers = main_parser.add_subparsers(help='sub-command help', dest='command')
+    # sub-commands of cromwell-tools
+    submit = subparsers.add_parser(
+            'submit', help='submit help', description='Submit a WDL workflow on Cromwell.')
+    wait = subparsers.add_parser(
+            'wait', help='wait help', description='Wait for one or more running workflow to finish.')
+    status = subparsers.add_parser(
+            'status', help='status help', description='Get the status of one or more workflows.')
+    health = subparsers.add_parser(
+            'health', help='health help',
+            description='Check that cromwell is running and that provided authentication is valid')
+
+    # cromwell url and authentication arguments apply to all sub-commands
+    cromwell_sub_commands = [submit, wait, status, health]
```

Comment:
The commands available through the CLI are missing "query", "abort" and "metadata". These could be added in a follow-up PR if you prefer since this refactor will already change a bunch of code!